### PR TITLE
Local docker test execution and linting

### PR DESCRIPTION
**Pull request by felixbb**
Uses caching of ubuntu and pip packages for getting results faster and saving bandwidth.

### CODE REVIEW

File: scripts/run_tests.sh [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+# limitations under the License.
+
+# Delete all running containers
+docker rm -f $(docker ps -a -q)
+
+docker build -t forseti/base -f scripts/docker/base . || exit 1
+docker build -t forseti/build -f scripts/docker/forseti --no-cache . || exit 1
+docker run -it -d --name build forseti/build /bin/bash || exit 1
+
+docker exec -it build /bin/sh -c "coverage run --source='google.cloud.security' --omit='__init__.py' -m unittest discover -s . -p '*_test.py'" || exit 1
```
Comment (mirons-google) [inline scripts/run_tests.sh]: We don't technically need a coverage run here unless if you're planning on having travis execute this?

File: scripts/run_tests.sh [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+docker exec -it build /bin/sh -c "pylint --rcfile=pylintrc --ignore-patterns='.*_test.*,.*test_.*,.*test.py,.*pb2.*' --output-format=text google" || exit 1
```
Comment (mirons-google) [inline scripts/run_tests.sh]: We have many of these pylint commands in various places and need to standardize on it's invocaiton.

Comment (felixbb) [inline scripts/run_tests.sh]: We should have no more than two - One in travis file and one here.

Comment (felixbb) [inline scripts/run_tests.sh]: If we want to have one place, we can make travis call this script but it'd be a little ugly and I don't want to touch the build process for now.

File: scripts/run_tests.sh (hunk 1 above)
Comment (felixbb) [inline scripts/run_tests.sh]: True, but it also runs the tests and I want to make sure the tests are ran exactly as in travis.

**Comment by blueandgold** [comment]
Adding `cla: yes` label, as the PR author is a googler at the time of this PR was created.

**Pull request merged**